public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -345,26 +347,39 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
 
   // By implementing getTagKeys and getTagValues we add ad-hoc filters functionality
   // Used in public/app/features/variables/adhoc/picker/AdHocFilterKey.tsx::fetchFilterKeys
-  getTagKeys(options: InfluxQuery) {
+  getTagKeys(options: DataSourceGetTagKeysOptions) {
+    let measurement = '';
+
+    if (options && options.filters) {
+      measurement = options.filters.find((f) => f.key === '__measurement__')?.value ?? '';
```

Comment:
I do not understand the `__measurement__` here. Where is it set initially?

Reply:
@itsmylife well, originally the code was looking for a `.measurement` prop that was passed in an object to this function (Grafana never passed anything but this function is also used from the QueryEditor), but I have not found anywhere in the InfluxDB query editor where it passes any argument to getTagKeys, but going to spin up InfluxDB and do some testing, and possibly remove this if the editor never passes measurement to it

Reply:
I did the refactoring of this part. (I must admit that it was not the best one)
The old code related to that call is here: 

- [getTagKeys](https://github.com/grafana/grafana/blob/4446d500a4ebfec0cd94886e2dcf0c03a0e7c2a3/public/app/plugins/datasource/influxdb/datasource.ts#L549-L553)
- [The place where the query is built ](https://github.com/grafana/grafana/blob/4446d500a4ebfec0cd94886e2dcf0c03a0e7c2a3/public/app/plugins/datasource/influxdb/query_builder.ts#L39-L42)

I think it is not being used and we can remove it.